Bio/Graphics/GenomeDiagram/_FeatureSet.py

```diff
@@ -96,12 +96,12 @@ def set_all_features(self, attr, value):
         Set the passed attribute of all features in the set to the
         passed value.
         """
-        changed = 0
         for feature in self.features.values():
-            # If the feature has the attribute, and the value should change
-            if hasattr(feature, attr):
-                if getattr(feature, attr) != value:
-                    setattr(feature, attr, value)  # set it to the passed value
+            try:
+                # If the feature has the attribute, set it to the passed value
+                setattr(feature, attr, value)
+            except AttributeError:
+                pass
```

Comment:
I can see no obvious reason for the original code here checking if the value was different, versus just setting it as you do. Should be fine.

Reply:
I think what I was trying to do was avoid a situation where an inappropriate attribute was set on a feature. For instance, there's not much to be gained setting `EC_number` for a `sig_peptide` feature.

I think the `try:except` approach makes sense, but if we don't check whether the feature already has the attribute, using `setattr()` will introduce the attribute where it did not exist before. That's a change of behaviour and might not be what you wanted to do.

I think:

```
 try:
    if hasattr(feature, attr):
        setattr(feature, attr, value)
```

Should keep the original behaviour.

Reply:
`setattr` raises an `AttributeError` if the attribute doesn't exist

Reply:
therefore we don't need to check `hasattr` first

Reply:
I'm not sure that's correct:

```python
Python 3.7.3 (default, Mar 27 2019, 16:54:48) 
[Clang 4.0.1 (tags/RELEASE_401/final)] :: Anaconda, Inc. on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> class Test():
...     def __init__(self):
...             self.a = 1
... 
>>> t = Test()
>>> t.a
1
>>> t.b
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Test' object has no attribute 'b'
>>> setattr(t, "b", 2)
>>> t.b
2
>>> t.__dict__
{'a': 1, 'b': 2}
```

Reply:
I see; thanks for checking. I was trying this with a built-in class.